packages/gatsby-plugin-sharp/src/gatsby-node.js

```diff
@@ -104,9 +109,18 @@ exports.onPostBootstrap = async ({ reporter, cache, store }) => {
   }
 }
 
+if (coreSupportsOnPluginInit) {
+  exports.unstable_onPluginInit = async ({ actions }, pluginOptions) => {
+    setActions(actions)
+    setPluginOptions(pluginOptions)
+  }
+}
+
 exports.onPreBootstrap = async ({ actions, emitter, cache }, pluginOptions) => {
-  setActions(actions)
-  setPluginOptions(pluginOptions)
+  if (!coreSupportsOnPluginInit) {
```

Comment:
probably can drop this check here and let it call `setActions` / `setPluginOptions` - this was done mainly to verify other code path works in practice

Reply:
Is there a chance that people have combinations of packages where `isGatsbyNodeLifecycleSupported` itself errors because it doesn't exist yet? So update to date sharp plugin but outdated utils

Reply:
In this PR `gatsby-plugin-sharp` add explicit dependency on `gatsby-plugin-utils`. So previous versions of sharp plugin wouldn't use it, "new" ones would use it and also have dependency that make sure it uses version to export `isGatsbyNodeLifecycleSupported`. The thing I like about it, is that this will work up to `gatsby@2.13.41` which is when we added `apis.json` file to be published to npm